aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt

```diff
@@ -178,6 +179,17 @@ class S3ExpressIdentityProviderConfig(codegenContext: ClientCodegenContext) : Co
                     )
                 }
 
+                ServiceConfig.BuilderBuild -> {
+                    rustTemplate(
+                        """
+                        self.runtime_plugins.push(
```

Comment:
I think this is probably a better place to hook into the runtime plugins since the ones in config are intended to be provided by customers:
https://github.com/smithy-lang/smithy-rs/blob/5761d1fe12e1a9235f9332122b867800db5d879a/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt#L490

We would need to add a codegen customization hook in there, but we should probably have one anyway.

Reply:
I thought about that too (including the need for a codegen customization hook there), but had a question whether once we're in `base_client_runtime_plugins`, a disabling value from method `Config::disable_s3_express_session_auth` and that in a profile file are stored in separate places? If they are consolidated into a single value, then there isn't a way for the environment variable to tell where that disabling value originally came from (i.e. if it comes from the method, then the env variable should not override it but if comes from a profile file, it should override it).

Reply:
That said, whatever we can do in `BuilderBuild`, we can also do in `base_client_runtime_plugins` such as accessing `pub(crate) config: crate::config::FrozenLayer,`. Doing so in `base_client_runtime_plugins` feels it's more right thing to do. Let me pursue that approach.